File: kotlin/sum-of-multiples/src/main/kotlin/SumOfMultiples.kt
class SumOfMultiples {
    companion object {
        fun sum(multiples: Set<Int>, limit: Int): Int {
            val factors: MutableList<Int> = mutableListOf()
            for (i in 1..(limit - 1)) {
                multiples.map { multiple: Int ->
                    if (i >= multiple && i.rem(multiple) == 0) {
                        factors.add(i)
                    }
                }
            }

            return factors.distinct().sum()
        }
    }
}

File: solutions/csharp/sum-of-multiples/1/SumOfMultiples.cs
using System.Dynamic;
using System.Transactions;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        var factors = new HashSet<int>();

        foreach (var multiple in multiples.Where(m => m > 0))
        {
            var current = multiple;
            while (current < max)
            {
                factors.Add(current);

                current += multiple;

            }
        }
        return factors.Sum();
    }
}